File: an2_sem4/MPP/Homework/sapt56/java/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/AngajatService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Angajat;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.IAngajatRepo;

public class AngajatService extends AbstractService<Integer, Angajat, IAngajatRepo>
    implements IAngajatService {
    public AngajatService(IAngajatRepo repo) {
        super(repo);
    }

    @Override
    public void register(Angajat angajat) throws EntityRepoException {
        var username = angajat.getUsername();
        var password = Utils.computeSha256Hash(angajat.getPassword());
        var email = angajat.getEmail();
        add(new Angajat(username, password, email));
    }

    @Override
    public Angajat login(String username, String password) throws EntityRepoException {
        return getRepo().findByCredentials(username, Utils.computeSha256Hash(password));
    }
}

File: an2_sem4/MPP/Homework/sapt78/cs/FestivalSellpoint/FestivalSellpoint.Service/AngajatService.cs
using FestivalSellpoint.Domain;
using FestivalSellpoint.Repo;
using System.Linq;
using System.Security.Cryptography;
using System.Text;

namespace FestivalSellpoint.Service
{
    public class AngajatService : AbstractService<int, Angajat, IAngajatRepo>, IAngajatService
    {
        public AngajatService(IAngajatRepo repo) : base(repo)
        {
        }

        public void Register(Angajat angajat)
        {
            var username = angajat.Username;
            var password = Utils.ComputeSha256Hash(angajat.Password);
            var email = angajat.Email;
            Add(new Angajat(username, password, email));
        }

        public Angajat Login(string username, string password)
            => Repo.FindByCredentials(username, Utils.ComputeSha256Hash(password));
       
    }
}
